src/Adapter/Shop/Context.php

```diff
@@ -73,4 +73,56 @@ public function getShopContextGroupConstant()
     {
         return \ShopCore::CONTEXT_GROUP;
     }
+
+    /**
+     * Get if it's a GroupShop context
+     *
+     * @return bool
+     */
+    public function isShopGroupContext()
+    {
+        return \ShopCore::getContext() === $this->getShopContextGroupConstant();
+    }
+
+    /**
+     * Get the ShopCore class CONTEXT_SHOP constant value
+     *
+     * @return int
+     */
+    public function getShopContextConstant()
+    {
+        return \ShopCore::CONTEXT_SHOP;
+    }
+
+    /**
+     * Get if it's a Shop context
+     *
+     * @return bool
+     */
+    public function isShopContext()
+    {
+        return \ShopCore::getContext() === $this->getShopContextConstant();
```

Comment:
I believe this method doesn't require an additional method. 
You might want to compare the returned value with the constant below directly.